Sources/StreamChat/Workers/ChannelListUpdater.swift

```diff
@@ -14,27 +14,92 @@ class ChannelListUpdater: Worker {
     ///
     func update(
         channelListQuery: ChannelListQuery,
-        completion: ((Result<ChannelListPayload, Error>) -> Void)? = nil
+        completion: ((Result<[ChatChannel], Error>) -> Void)? = nil
     ) {
         fetch(channelListQuery: channelListQuery) { [weak self] in
             switch $0 {
             case let .success(channelListPayload):
-                self?.database.write { session in
-                    try session.saveChannelList(payload: channelListPayload, query: channelListQuery)
-                } completion: { error in
-                    if let error = error {
-                        log.error("Failed to save `ChannelListPayload` to the database. Error: \(error)")
-                        completion?(.failure(error))
-                    } else {
-                        completion?(.success(channelListPayload))
-                    }
-                }
+                self?.writeChannelListPayload(
+                    payload: channelListPayload,
+                    query: channelListQuery,
+                    completion: completion
+                )
             case let .failure(error):
                 completion?(.failure(error))
             }
         }
     }
-    
+
+    func resetChannelsQuery(
```

Comment:
I think it makes sense to write unit tests around this logic. We should have all the mocks available for it.

Reply:
Oh, yeah, that's for sure. As explained in the description, a whole PR just focusing on Unit Tests is coming